packages/cli/src/lifecycles/prerender.js

```diff
@@ -187,23 +187,23 @@ async function staticRenderCompilation(compilation) {
   console.info("pages to generate", `\n ${pages.map((page) => page.route).join("\n ")}`);
 
   await Promise.all(
-    pages.map(async (page) => {
-      const { route, outputHref } = page;
-      const scratchUrl = toScratchUrl(outputHref, context);
-      const url = new URL(`http://localhost:${config.port}${route}`);
-      const request = new Request(url);
-
-      let body = await (await servePage(url, request, plugins)).text();
-      body = await (await interceptPage(url, request, plugins, body)).text();
-
-      await trackResourcesForRoute(body, compilation, route);
-      await createOutputDirectory(route, new URL(scratchUrl.href.replace("index.html", "")));
-      await fs.writeFile(scratchUrl, body);
-
-      console.info("generated page...", route);
-
-      return Promise.resolve();
-    }),
+    pages.map((page) =>
```

Comment:
This seems like it could easily just be a `for ... of`

Reply:
likewise this would no longer run concurrently if we removed the promise.all.